File: app/src/main/java/com/incendiary/android/di/component/ApplicationComponent.java
package com.incendiary.android.di.component;

import android.app.Application;
import android.content.Context;
import dagger.Component;
import javax.inject.Singleton;
import com.incendiary.android.data.DataManager;
import com.incendiary.android.data.SyncService;
import com.incendiary.android.data.local.DatabaseHelper;
import com.incendiary.android.data.remote.ApiService;
import com.incendiary.android.di.ApplicationContext;
import com.incendiary.android.di.module.ApplicationModule;
import com.incendiary.android.di.module.NetworkModule;
import com.incendiary.android.util.RxEventBus;

@Singleton @Component(modules = {
    ApplicationModule.class, NetworkModule.class
}) public interface ApplicationComponent {

  void inject(SyncService syncService);

  @ApplicationContext Context context();

  Application application();
  ApiService apiService();
  DatabaseHelper databaseHelper();
  DataManager dataManager();
  RxEventBus eventBus();
}

File: app/src/main/java/com/incendiary/androidboilerplate/di/component/ApplicationComponent.kt
package com.incendiary.androidboilerplate.di.component

import android.app.Application
import android.content.Context
import com.incendiary.androidboilerplate.data.DataManager
import com.incendiary.androidboilerplate.data.SyncService
import com.incendiary.androidboilerplate.data.local.DatabaseHelper
import com.incendiary.androidboilerplate.data.remote.ApiService
import com.incendiary.androidboilerplate.di.ApplicationContext
import com.incendiary.androidboilerplate.di.module.ApplicationModule
import com.incendiary.androidboilerplate.di.module.NetworkModule
import com.incendiary.androidboilerplate.util.RxEventBus
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, NetworkModule::class))
interface ApplicationComponent {

	fun inject(syncService: SyncService)

	@ApplicationContext fun context(): Context

	fun application(): Application
	fun apiService(): ApiService
	fun databaseHelper(): DatabaseHelper
	fun dataManager(): DataManager
	fun eventBus(): RxEventBus
}
